sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceJobPart.cs

```diff
@@ -483,7 +503,11 @@ internal void DisposeHandlers()
 
         private async Task<StorageResourceCopyFromUriOptions> GetCopyFromUriOptionsAsync(CancellationToken cancellationToken)
         {
-            StorageResourceCopyFromUriOptions options = default;
+            StorageResourceItemProperties properties = await _sourceResource.GetPropertiesAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This will always be cached in the resource from initial call when initializing the job part right? If fo, maybe add a comment so we remember that in the future.

Reply:
Yes it will be. This is done on line 186 of this same file. I checked where this method was called previously, which should cache the properties on the source resource.